src/scenic/syntax/compiler.py

```diff
@@ -171,3 +172,158 @@ def visit_ApparentlyFacingSpecifier(self, node: s.ApparentlyFacingSpecifier):
             if node.base is not None
             else [],
         )
+
+    # Operators
+
+    def visit_RelativePositionOp(self, node: s.RelativePositionOp):
+        return ast.Call(
+            func=ast.Name(id="RelativePosition", ctx=loadCtx),
+            args=[self.visit(node.target)],
+            keywords=[]
+            if node.base is None
+            else [ast.keyword(arg="Y", value=self.visit(node.base))],
+        )
+
+    def visit_RelativeHeadingOp(self, node: s.RelativeHeadingOp):
+        return ast.Call(
+            func=ast.Name(id="RelativeHeading", ctx=loadCtx),
+            args=[self.visit(node.target)],
+            keywords=[]
+            if node.base is None
+            else [ast.keyword(arg="Y", value=self.visit(node.base))],
+        )
+
+    def visit_ApparentHeadingOp(self, node: s.ApparentHeadingOp):
+        return ast.Call(
+            func=ast.Name(id="ApparentHeading", ctx=loadCtx),
+            args=[self.visit(node.target)],
+            keywords=[]
+            if node.base is None
+            else [ast.keyword(arg="Y", value=self.visit(node.base))],
+        )
+
+    def visit_DistanceFromOp(self, node: s.DistanceFromOp):
+        assert node.target is not None or node.base is not None
+        target = ego if node.target is None else self.visit(node.target)
+        base = ego if node.base is None else self.visit(node.base)
```

Comment:
If `ego` is still a function in the veneer, don't these need to be calls rather than just names?

In any case, for most of the operators which use ego implicitly, you used the approach of passing them an optional keyword argument when specified. Why not do that here as well? The `DistanceFrom(X, Y)` function in the veneer has a default value of `None` for `Y`, which it then replaces with the ego.

Reply:
You're right that the default operands should be defined inside the veneer. I changed it so that 

- if there is only one operand (either `from` or `to`), use it as `X`
- if there are two operands (both `from` and `to`), use `to` as `X` and `from` as `Y`

b78f733d00cdfbac0f9f251d90a329c719d3153a